### PR TITLE
wpa-supplicant: dynamic config

### PR DESCRIPTION
**Pull request by sknsean**

### CODE REVIEW

**Comment by sknsean** [comment]
I can't find where the dependency for libgpg-eror is and packageqa showed nothing, so i have removed it

**Comment by esben** [comment]
I am not deeply into wpa_suplicant currently, but from an OE-lite perspective: :+1:

File: recipes/wpa-supplicant/wpa-supplicant.inc [hunk 1]
```diff
@@ -9,17 +9,18 @@ inherit make c++ pkgconfig
 
 DEPENDS_LIBC = "libdl"
 DEPENDS_LIBC:HOST_LIBC_mingw = ""
-DEPENDS += "libdbus-1 libnl-3 libnl-genl-3 libnl-genl-3 libgnutls libgcrypt \
-	libgpg-error ${DEPENDS_LIBC}"
+DEPENDS += "libnl-3 libnl-genl-3 libnl-genl-3 librt \
+	    ${CRYPTO} ${DEPENDS_LIBC}"
```
Comment (kimrhh) [inline recipes/wpa-supplicant/wpa-supplicant.inc]: why are you removing dbus? this is positively used in some setups

File: recipes/wpa-supplicant/wpa-supplicant.inc [hunk 2]
```diff
@@ -9,17 +9,18 @@ inherit make c++ pkgconfig
...
-           file://defconfig-gnutls \
            file://wpa-supplicant.sh \
            file://wpa_supplicant.conf \
            file://wpa_supplicant.conf-sane \
 	   "
 
-S = "${SRCDIR}/wpa_supplicant-${PV}/wpa_supplicant"
+S = "${SRCDIR}/wpa_supplicant-${PV}"
+
+WPADIR = "wpa_supplicant"
```
Comment (kimrhh) [inline recipes/wpa-supplicant/wpa-supplicant.inc]: upstream tarball just unpacks to unversioned dir?

Comment (kimrhh) [inline recipes/wpa-supplicant/wpa-supplicant.inc]: core/conf/oelayout.conf:S                = "${SRCDIR}/${P}"
do you really need to override ${S} ?

Comment (hundeboll) [inline recipes/wpa-supplicant/wpa-supplicant.inc]: - vs _ ?

File: recipes/wpa-supplicant/wpa-supplicant/wpa_supplicant-2.5-fix-libressl.patch [hunk 3]
```diff
@@ -0,0 +1,65 @@
+--- wpa_supplicant-2.5/src/crypto/tls_openssl.c.orig	2016-01-23 14:53:34.479639958 +0000
```
Comment (hundeboll) [inline recipes/wpa-supplicant/wpa-supplicant/wpa_supplicant-2.5-fix-libressl.patch]: Please add a comment on where you found this patch, why it is needed and the upstream status (should it be reported?)...

File: recipes/wpa-supplicant/wpa-supplicant.inc (hunk 2 above)
Comment (sknsean) [inline recipes/wpa-supplicant/wpa-supplicant.inc]: @hundeboll  :+1:

**Pull request merged**